reactor-core/src/main/java/reactor/core/publisher/FluxOnBackpressureBuffer.java

```diff
@@ -148,10 +156,8 @@ public void onNext(T t) {
 				Operators.onNextDropped(t, ctx);
 				return;
 			}
-			if (!queue.offer(t)) {
-				Throwable ex =
-						Operators.onOperatorError(s, Exceptions.failWithOverflow(), t,
-								ctx);
+			if (queue.size() >= capacityOrSkip || !queue.offer(t)) {
```

Comment:
Maybe could be optimized a bit, like checking the size ONLY if the capacityOrSkip is != Integer MAX.

Reply:
good point, fixed. also added `scan(CAPACITY)` support